src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let currentState = Object.assign({}, state);
+
+  for (const i in actions) {
+    switch (actions[i].type) {
```

Comment:
Fix the variable name and use destructuring
```suggestion
  for (const action in actions) {
    const { type, ....... } = action;
    
    switch (type) {
```